autoload/codefmt.vim

```diff
@@ -156,7 +156,67 @@ if !exists('s:gofmt')
   call codefmtlib#AddDefaultFormatter(s:gofmt)
 endif
 
+" Formatter: autopep8
+if !exists('s:autopep8')
+  let s:autopep8 = {
+      \ 'name': 'autopep8',
+      \ 'setup_instructions': 'Install autopep8 and '}
+
+  function s:autopep8.IsAvailable() abort
+    return executable(s:plugin.Flag('autopep8_executable'))
+  endfunction
+
+  function s:autopep8.AppliesToBuffer() abort
+    return &filetype is# 'python'
+  endfunction
+
+
+  ""
+  " Reformat the current buffer with autopep8 or the binary named in
+  " @flag(autopep8_executable), only targeting the range between {startline} and
+  " {endline}.
+  function s:autopep8.FormatRange(startline, endline) abort
+    " Hack range formatting by formatting range individually, ignoring context.
+    let l:cmd = [ s:plugin.Flag('autopep8_executable'), "-"  ]
+    call maktaba#ensure#IsNumber(a:startline)
+    call maktaba#ensure#IsNumber(a:endline)
+    let l:lines = getline(1, line('$'))
+    let l:input = join(l:lines[a:startline - 1 : a:endline - 1], "\n")
```

Comment:
btw, while i gonna store cached value in Flag, then it will be possible to set it manually

or Flags are not suitable for such purpose and intended to be _only_ for storing some default settings, but not any variable data?